### PR TITLE
Documentation Formatting

### LINKED ISSUE #116: Documentation formatting
Use `mdl` for formatting markdown documentation.

See https://github.com/DanNixon/satori/commit/e71b2b3835f60d26cf389a42bf73b6a94da9a2f3 for an example of how this can be done.

Start with only this in the rules file:

```ruby
all
exclude_rule 'MD013'
```

### PR DESCRIPTION
**Pull request by Henry-Denny**
Closes #116

### CODE REVIEW

File: treefmt.toml [hunk 1]
```diff
@@ -6,3 +6,7 @@ includes = [ "*.nix" ]
 command = "rustfmt"
 options = ["--edition", "2021"]
 includes = [ "*.rs" ]
+
+[formatter.markdown]
```
Comment (DanNixon) [inline treefmt.toml:10]: A small thing, but could the formatters be kept in alphabetical order by filetype.

File: trace-to-events/README.md [hunk 2]
```diff
@@ -1,42 +1,52 @@
...
+- `AdvancedMuonDetector`:        Detects events using differential discriminators. Event lists consist of time and voltage values.
+- `help`:         Print this message or the help of the given subcommand(s)
+
+### Constant Phase Discriminator
 
-### Constant Phase Discriminator:
 `trace-to-events --broker <BROKER> constant-phase-discriminator --threshold-trigger <THRESHOLD_TRIGGER>`
 
-```
+```shell
```
Comment (DanNixon) [inline trace-to-events/README.md:36]: This code block should just be removed, it seems to serve no purpose.

File: trace-to-events/README.md [hunk 3]
```diff
@@ -56,26 +66,32 @@ A threshold is given by a triple of the form "threshold,duration,cool_down", thr
 ```
 
 ## Configuring the Detector Pipeline
+
 Given an iterator of type u16 (aliased as Intensity in the crate), the pipeline is setup as follows:
-```
+
+```rust
 // trace : Vec<u16>
     let raw = trace
```
Comment (DanNixon) [inline trace-to-events/README.md:74]: Could the extra indentation be removed here.

File: run-simulator/README.md [hunk 4]
```diff
@@ -2,37 +2,48 @@
...
+```shell
 run-simulator --help
 ```
 
 ## Time Format
+
 If  the ``time`` command line argument is ommitted then the current time is used. This argument expects a date/time given in the following format:
-```
+
+```shell
```
Comment (DanNixon) [inline run-simulator/README.md:25]: These can be replaced with inline code spans and inlined with the above text.

File: run-simulator/README.md [hunk 5]
```diff
@@ -2,37 +2,48 @@
...
 [YYYY]-[mm]-[dd]T[HH]:[MM]:[SS].[nnnnnnnnn]Z
 ```
+
 That is the hyphen-separated date followed by colon-separated time terminated by "Z". Between the date and time can go either a space, or a "T".
 
 If a space is used the timestamp must be enclosed by quoatation marks.
 
 For instance:
```
Comment (DanNixon) [inline run-simulator/README.md:39]: As can these.

File: docs/developer_setup.md [hunk 6]
```diff
@@ -6,10 +6,10 @@ WSL2 should probably work, but has not been extensively tested.
...
 
 1. Install [Nix](https://nixos.org/) using the [Determinate Installer](https://github.com/DeterminateSystems/nix-installer#usage).
-2. Install [direnv](https://direnv.net/docs/installation.html).
-3. `git clone https://github.com/STFC-ICD-Research-and-Design/supermusr-data-pipeline`
-4. `cd supermusr-data-pipeline`
-5. `direnv allow`
+1. Install [direnv](https://direnv.net/docs/installation.html).
+1. `git clone https://github.com/STFC-ICD-Research-and-Design/supermusr-data-pipeline`
+1. `cd supermusr-data-pipeline`
+1. `direnv allow`
```
Comment (DanNixon) [inline docs/developer_setup.md:12]: Was the issue here the fact that some documents were auto numbered and some not?

If so I would prefer explicit numbering throughout.

**Review by DanNixon — CHANGES_REQUESTED** [review]

File: docs/developer_setup.md (hunk 6 above)
Comment (Henry-Denny) [reply docs/developer_setup.md:12]: The default ordered list style in [this rule](https://github.com/markdownlint/markdownlint/blob/main/docs/RULES.md#md029---ordered-list-item-prefix) is 'one'. I have reverted to the original numbering style and have updated the style to 'ordered' inside `.mdl_style.rb`.

File: trace-archiver/README.md [hunk 7]
```diff
@@ -7,7 +7,8 @@ A file is created for each received trace message and saved in a file named
 using the format `frame_{timestamp}_{digitizer_id}_{frame_number}.h5`.
 
 The structure of the HDF5 file is as follows:
-```
+
+```shell
```
Comment (DanNixon) [inline trace-archiver/README.md:11]:
```suggestion
```text
```

**Review by DanNixon — CHANGES_REQUESTED** [review]

**Pull request merged**